OpenSim/Common/Component.h

```diff
@@ -1554,6 +1554,7 @@ template <class T> friend class ComponentMeasure;
             _connectorsTable[connector.getName()] = ix;
         }
     }
+protected:
```

Comment:
> I am also thinking we could rename the method to createSubcomponentsTraversalTree()

Did we already rename other methods to use 'sub', like `addComponent`? 



> Also add doxygen comments especially if new "environment" or "study" like Components should call this method as part of the their extendConnect() implementation (e.g. TheWorld component in testComponentInterface()).

What do you think about putting the comment in `extendConnect()`, since the implementor of `TheWorld` won't know to look for this method in order to learn that they should use it.

Also, what if a top level component, e.g. Model, is nested within another top level component, like TheWorld or Environment? Model will need to call `initComponentTree...()` because it wants to stand alone. But `TheWorld` will also call it, right? This discussion might not belong in this PR.